### PR TITLE
Fix Publish Test Results path logic for Release Management

### PR DESCRIPTION
**Pull request by nigurr**
* **agent.buildDirectory** is not supported in RM Definitions.
https://www.visualstudio.com/docs/release/author-release-definition/understanding-tasks#predefvariables

### CODE REVIEW

File: Tasks/Maven/maventask.ts [hunk 1]
```diff
@@ -186,7 +186,7 @@ function publishJUnitTestResults(testResultsFiles: string) {
     // Check for pattern in testResultsFiles
     if (testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
         tl.debug('Pattern found in testResultsFiles parameter');
-        var buildFolder = tl.getVariable('agent.buildDirectory');
+        var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
```
Comment (AshwiniChalla) [inline Tasks/Maven/maventask.ts]: This wil lwork for build as well ?

Comment (AshwiniChalla) [inline Tasks/Maven/maventask.ts]: can we have a null check after retreiving the build folder ?

Comment (nigurr) [inline Tasks/Maven/maventask.ts]: yes.

File: Tasks/Gradle/gradletask.ts [hunk 2]
```diff
@@ -88,7 +88,7 @@ function publishTestResults(publishJUnitResults, testResultsFiles: string) {
         //check for pattern in testResultsFiles
         if (testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
             tl.debug('Pattern found in testResultsFiles parameter');
-            var buildFolder = tl.getVariable('agent.buildDirectory');
+            var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
```
Comment (AshwiniChalla) [inline Tasks/Gradle/gradletask.ts]: no test changes?

File: Tasks/PublishTestResults/publishtestresults.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ tl.debug('publishRunAttachments: ' + publishRunAttachments);
 //check for pattern in testResultsFiles
 if(testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
   tl.debug('Pattern found in testResultsFiles parameter');
-  var buildFolder = tl.getVariable('agent.buildDirectory');
+  var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
```
Comment (AshwiniChalla) [inline Tasks/PublishTestResults/publishtestresults.ts]: Can you scan through all our tasks for this?

File: Tasks/Gradle/gradletask.ts (hunk 2 above)
Comment (prawalagarwal) [inline Tasks/Gradle/gradletask.ts]: +1

**Comment by prawalagarwal** [comment]
Signed off with comments

File: Tasks/Maven/maventask.ts (hunk 1 above)
Comment (nigurr) [inline Tasks/Maven/maventask.ts]: These are pre-defined variables. Agent wouldn't run without these

File: Tasks/Gradle/gradletask.ts (hunk 2 above)
Comment (nigurr) [inline Tasks/Gradle/gradletask.ts]: Presently there are no L0 tests with Gradle + Publish Test

File: Tasks/Maven/maventask.ts (hunk 1 above)
Comment (AshwiniChalla) [inline Tasks/Maven/maventask.ts]: Hope you have done manual testing . Please test with coreclr agent too.

File: Tasks/PublishTestResults/publishtestresults.ts (hunk 3 above)
Comment (nigurr) [inline Tasks/PublishTestResults/publishtestresults.ts]: yes. tested

File: Tasks/Maven/maventask.ts (hunk 1 above)
Comment (nigurr) [inline Tasks/Maven/maventask.ts]: yes. tested

File: Tasks/Gradle/gradletask.ts (hunk 2 above)
Comment (AshwiniChalla) [inline Tasks/Gradle/gradletask.ts]: Can you please add tests for those also.

**Comment by AshwiniChalla** [comment]
Add tests and then consider signoff from myside.

**Pull request merged**